sklearn/multioutput.py

```diff
@@ -365,3 +368,232 @@ def score(self, X, y):
+            self.order_ = np.array(range(Y.shape[1]))
+        elif isinstance(self.order_, str):
+            if self.order_ == 'random':
+                self.order_ = random_state.permutation(Y.shape[1])
+        elif sorted(self.order_) != list(range(Y.shape[1])):
+                raise ValueError("invalid order")
+
+        self.estimators_ = [clone(self.base_estimator)
+                            for _ in range(Y.shape[1])]
+
+        self.classes_ = []
+
+        if self.cv is None:
+            Y_pred_chain = Y[:, self.order_]
+            if sp.issparse(X):
+                X_aug = sp.hstack((X, Y_pred_chain), format='lil')
+                X_aug = X_aug.tocsr()
+            else:
+                X_aug = np.hstack((X, Y_pred_chain))
+
+        elif sp.issparse(X):
+            Y_pred_chain = sp.lil_matrix((X.shape[0], Y.shape[1]))
+            X_aug = sp.hstack((X, Y_pred_chain), format='lil')
+
+        else:
+            Y_pred_chain = np.zeros((X.shape[0], Y.shape[1]))
+            X_aug = np.hstack((X, Y_pred_chain))
+
+        del Y_pred_chain
+
+        for chain_idx, estimator in enumerate(self.estimators_):
+            y = Y[:, self.order_[chain_idx]]
+            estimator.fit(X_aug[:, :(X.shape[1] + chain_idx)], y)
+            if self.cv is not None and chain_idx < len(self.estimators_) - 1:
+                col_idx = X.shape[1] + chain_idx
+                cv_result = cross_val_predict(
+                    self.base_estimator, X_aug[:, :col_idx],
+                    y=y, cv=self.cv)
+                if sp.issparse(X_aug):
+                    X_aug[:, col_idx] = np.expand_dims(cv_result, 1)
+                else:
+                    X_aug[:, col_idx] = cv_result
+
+            self.classes_.append(estimator.classes_)
+        return self
+
+    def predict(self, X):
+        """Predict on the data matrix X using the ClassifierChain model.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+
+        Returns
+        -------
+        Y_pred : array-like, shape (n_samples, n_classes)
+        """
+        X = check_array(X, accept_sparse=True)
+        Y_pred_chain = np.zeros((X.shape[0], len(self.estimators_)))
+        for chain_idx, estimator in enumerate(self.estimators_):
+            previous_predictions = Y_pred_chain[:, :chain_idx]
+            if sp.issparse(X):
+                if chain_idx == 0:
+                    X_aug = X
+                else:
+                    X_aug = sp.hstack((X, previous_predictions))
+            else:
+                X_aug = np.hstack((X, previous_predictions))
+            Y_pred_chain[:, chain_idx] = estimator.predict(X_aug)
+
+        inv_order = np.empty_like(self.order_)
+        inv_order[self.order_] = np.arange(len(self.order_))
+        Y_pred = Y_pred_chain[:, inv_order]
+
+        return Y_pred
+
+    @if_delegate_has_method('base_estimator')
+    def predict_proba(self, X):
```

Comment:
I think that this could use a little bit more documentation. Specifically, you may want to clarify that `predict` and `predict_proba` don't change the input to the later models in the chain, it always uses `predict.`